app/api/v1/endpoints/auth.py

```diff
@@ -11,7 +11,7 @@
 @router.post("/register", response_model=UserInDB)
 @limiter.limit("50/minute")
 def register_user_endpoint(
-    request: Request, user_in: UserCreate, db: Session = Depends(deps.get_db)
+    request: Request, user_in: UserCreateUpdate, db: Session = Depends(deps.get_db)
```

Comment:
this approach not fine to pass the db session from router to the crud method, we can create a session within the crud method. Follow the same approach for all the endpoint.